lib/suse/connect/zypper.rb

```diff
@@ -65,6 +65,14 @@ def remove_service(service_name)
           remove_service_credentials(service_name)
         end
 
+        # @param product identifier [String]
+        # Returns an array of hashes of all solvable products
+        def find_products(identifier)
+          zypper_out = call("--xmlout --non-interactive search -s -t product #{identifier}", false)
+          xml_doc = REXML::Document.new(zypper_out, compress_whitespace: [])
+          xml_doc.elements.each('stream/search-result/solvable-list/solvable') {}.map(&:to_hash)
```

Comment:
IMO notation below
* is way easier to grasp and you don't need to skip yielding
* does not use iterator which is not needed here

```ruby
xml_doc.elements.to_a('stream/search-result/solvable-list/solvable').map(&:to_hash)
```